examples/text-classification/run_pl_glue.py

```diff
@@ -118,78 +208,129 @@ def _eval_end(self, outputs):
-        return {"avg_test_loss": logs["val_loss"], "log": logs, "progress_bar": logs}
+        return to_return
 
     @staticmethod
-    def add_model_specific_args(parser, root_dir):
-        # Add NER specific options
-        BaseTransformer.add_model_specific_args(parser, root_dir)
+    def add_model_specific_args(parent_parser):
+        parser = ArgumentParser(parents=[parent_parser], add_help=False)
         parser.add_argument(
             "--max_seq_length",
             default=128,
             type=int,
             help="The maximum total input sequence length after tokenization. Sequences longer "
             "than this will be truncated, sequences shorter will be padded.",
         )
-
         parser.add_argument(
-            "--task", default="", type=str, required=True, help="The GLUE task to run",
+            "--task", type=str, required=True, help="The GLUE task to run",
         )
-
         parser.add_argument(
-            "--data_dir",
-            default=None,
-            type=str,
-            required=True,
-            help="The input data dir. Should contain the training files for the CoNLL-2003 NER task.",
+            "--data_dir", default="./glue_dir", type=str, help="Directory to save/load processed cache data"
         )
-
         parser.add_argument(
             "--overwrite_cache", action="store_true", help="Overwrite the cached training and evaluation sets"
         )
-
         return parser
 
 
-if __name__ == "__main__":
-    parser = argparse.ArgumentParser()
-    add_generic_args(parser, os.getcwd())
-    parser = GLUETransformer.add_model_specific_args(parser, os.getcwd())
-    args = parser.parse_args()
+def parse_args(args=None):
+    parser = ArgumentParser()
+
+    # add some script specific args
+    parser.add_argument("--seed", type=int, default=42)
+    parser.add_argument(
+        "--output_dir", type=str, default="", help="Directory to write outputs to or load checkpoint from"
+    )
+    parser.add_argument("--do_train", action="store_true", help="Run training loop")
+    parser.add_argument("--do_predict", action="store_true", help="Run test loop")
+
+    # enable all trainer args
+    parser = Trainer.add_argparse_args(parser)
+
+    # add the base module args
+    parser = BaseTransformer.add_model_specific_args(parser)
 
-    # If output_dir not provided, a folder will be generated in pwd
-    if args.output_dir is None:
-        args.output_dir = os.path.join("./results", f"{args.task}_{time.strftime('%Y%m%d_%H%M%S')}",)
-        os.makedirs(args.output_dir)
+    # add the glue module args
+    parser = GLUETransformer.add_model_specific_args(parser)
 
+    # cook them all up :)
+    args = parser.parse_args(args)
+
+    return args
+
+
+def main(args):
+
+    # Init Model
     model = GLUETransformer(args)
-    trainer = generic_train(model, args)
 
-    # Optionally, predict on dev set and write to output_dir
+    # Init Trainer
+    trainer = Trainer.from_argparse_args(args)
```

Comment:
can we use `generic_train` or no?

Reply:
I got rid of it. Instead, we use the built in `Trainer` argparser (and the base model/subclass argparsers) to handle everything `generic_train` was doing before.